Code/Source/RobotImporter/RobotImporterWidget.cpp

```diff
@@ -63,20 +76,23 @@ namespace ROS2
                     ReportError("User cancelled");
                     return;
                 }
-                RobotImporter::Import(
-                    { urdfPath.value(), prefabPath.value() },
-                    [this](const AZStd::string& message)
-                    {
-                        ReportInfo(message);
-                    },
-                    [this](const AZStd::string& message)
-                    {
-                        ReportError(message);
-                    });
+                m_robotImporter.Import({ urdfPath.value(), prefabPath.value() });
+                m_selectFileButton->setEnabled(false);
+                m_importerUpdateTimer->start(500);
             });
         setLayout(mainLayout);
     }
 
+    void RobotImporterWidget::ImporterTimerUpdate()
+    {
+        m_robotImporter.Update(
+            [this]()
+            {
+                m_importerUpdateTimer->stop();
+                m_selectFileButton->setEnabled(true);
```

Comment:
Could we simply set this button status more directly on timer elapse?

Reply:
Not really: https://github.com/RobotecAI/o3de-ros2-gem/pull/142#discussion_r954939608